feat: Spanish localization

IanKrieger opened the pull request:
Spanish translation for: https://github.com/brave/roadmap/issues/1115

---

<img width="727" alt="Screen Shot 2024-03-18 at 7 55 41 AM" src="https://github.com/brave/ads-ui/assets/48930920/941fc962-4789-4170-b4c5-f00cf1812bf0">

iambrianfung commented on src/user/analytics/analyticsOverview/components/MetricSelect.tsx:
```diff
@@ -43,7 +43,7 @@ export default function MetricSelect({
       value: "landings",
       label: msg`Site visits`,
-      tooltip: msg`Counted if the user clicks an ad and spends at least 10 seconds on the advertiser's website, with the website open in an active browser tab. The 10 seconds must be spent on the site after arriving by clicking the ad link, and the tab must remain open and active the entire time for the visit to count.`,
+      tooltip: msg`Counted if the user clicks an ad and spends at least 5 seconds on the advertiser's website, with the website open in an active browser tab. The 5 seconds must be spent on the site after arriving by clicking the ad link, and the tab must remain open and active the entire time for the visit to count.`,
```
we're not at 5s yet.

iambrianfung replied:
a couple instances of this

IanKrieger commented:
> But how do I test this? I'd like to see an easy way to bring up the dashboard in spanish so I can review what English test is still showing.

I changed brave browser language beta to the primary language to be in spanish, as seen in the screenshot I added. Will have to deploy to make sure behaviour is the same,

> Also, as we change the english strings later, what breaks / reports to give us feedback that we now need additional strings translated?

That should be handled when you run `npm run extract` any changes to text are matched against an exact id which in this case is basically the text. if that id doesn't exist, it says a translation is missing.

tackley commented:
> > But how do I test this? I'd like to see an easy way to bring up the dashboard in spanish so I can review what English test is still showing.
> 
> I changed brave beta browser's primary language to be in spanish, as seen in the screenshot I added. Will have to deploy to make sure behaviour is the same,

OK!  In a future PR I'd like us to make trying different locales easier (perhaps also including the [pseudolocale](https://lingui.dev/guides/pseudolocalization)) for testing. Either by 1) supporting a `locale=en-US` query parameter on application load to force the locale detection or 2) (less preferred) a dev time env var e.g. `OVERRIDE_LOCALE=en-US npm start` or 3) (even less preferred) writing up in the readme how to switch locales in the brave. But not as part of this PR.

I didn't see the screenshot when I first wrote the comment ;)

> 
> > Also, as we change the english strings later, what breaks / reports to give us feedback that we now need additional strings translated?
> 
> That should be handled when you run `npm run extract` any changes to text are matched against an exact id which in this case is basically the text. if that id doesn't exist, it says a translation is missing.

Ahh, yes, that makes sense. I think in due course we'll lose track, and I'd like a build-time check / report if possible - especially if the values have changed - but again let's hold off on that for now.
